quodlibet/quodlibet/ext/playlist/export_to_folder.py

```diff
@@ -0,0 +1,165 @@
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation
+
+from gi.repository import Gtk
+
+from quodlibet import _
+from quodlibet.pattern import FileFromPattern
+from quodlibet.plugins import PluginConfig, ConfProp
+from quodlibet.plugins.playlist import PlaylistPlugin
+from quodlibet.qltk import Icons
+from quodlibet.qltk.notif import Task
+from quodlibet.qltk.window import Dialog
+from quodlibet.util import copool
+from quodlibet.util.dprint import print_d
+
+from shutil import copyfile
+
+
+class ExportToFolderDialog(Dialog):
+    """A dialog to collect export settings"""
+
+    def __init__(self, parent, pattern):
+        super(ExportToFolderDialog, self).__init__(
+            title=_("Export Playlist to Folder"),
+            transient_for=parent, use_header_bar=True)
+
+        self.set_default_size(400, -1)
+        self.set_resizable(True)
+        self.set_border_width(6)
+        self.vbox.set_spacing(6)
+
+        self.add_button(_("_Cancel"), Gtk.ResponseType.CANCEL)
+        self.add_button(_("_Export"), Gtk.ResponseType.OK)
+        self.set_default_response(Gtk.ResponseType.OK)
+
+        box = Gtk.VBox(spacing=6)
+
+        destination_label = Gtk.Label(_("Destination folder:"))
+        destination_label.set_line_wrap(True)
+        destination_label.set_xalign(0.0)
+        box.pack_start(destination_label, True, True, 0)
+
+        frame = Gtk.Frame()
+        self.directory_chooser = \
+            Gtk.FileChooserWidget(action=Gtk.FileChooserAction.SELECT_FOLDER)
+        self.directory_chooser.set_select_multiple(False)
+        self.directory_chooser.set_border_width(1)
+        frame.add(self.directory_chooser)
+        frame.set_shadow_type(Gtk.ShadowType.IN)
+        frame.set_border_width(0)
+        box.pack_start(frame, True, True, 0)
+
+        pattern_label = Gtk.Label(_("Filename pattern:"))
+        pattern_label.set_line_wrap(True)
+        pattern_label.set_xalign(0.0)
+        box.pack_start(pattern_label, True, True, 0)
+
+        self.pattern_entry = Gtk.Entry(text=pattern)
+        box.pack_start(self.pattern_entry, True, True, 0)
+
+        self.vbox.pack_start(box, True, True, 0)
+
+        self.set_response_sensitive(Gtk.ResponseType.OK, False)
+
+        def changed(*args):
+            has_directory = self.directory_chooser.get_filename() is not None
+            self.set_response_sensitive(Gtk.ResponseType.OK, has_directory)
+
+            pattern_text = self.pattern_entry.get_text()
+            has_pattern = pattern_text != ""
+            self.set_response_sensitive(Gtk.ResponseType.OK, has_pattern)
+
+        self.directory_chooser.connect("selection-changed", changed)
+        self.pattern_entry.connect("changed", changed)
+
+        self.get_child().show_all()
+
+
+class Config(object):
+    _config = PluginConfig(__name__)
+
+    DEFAULT_PATTERN = "<artist> - <title>"
```

Comment:
I recommend perhaps using a tied tag "<artist~title>" as it works better if is either is missing (e.g. in Classical music `0004. -  - Sonata K.380.flac`")
...unless those filenames themselves needs parsing by something strict I guess, but this is probably not the _default_ case...